### PR TITLE
Release 2.6.0

### PR DESCRIPTION
**Pull request by jenia81**
- Add support for an external signing tool
   - The feature enables to use of an external signing tool for signing the manifest. 
   - Add a new CLI option `--signing-tool` to the `manifest-dev-tool init` command.
   - The `--key` CLI option of the `manifest-dev-tool init` command can receive an identifier private key 
     or a private key file name that will be used to sign the manifest with the external signing tool provided.
   - Add a new configuration option `signing-tool` that is used by the `manifest-tool create` command.
   - The `--key` CLI option of the `manifest-tool create` or `manifest-tool create-v1` command can receive 
     an identifier private key or a private key file name that will be used to sign the manifest with 
     the external signing tool provided.
   - Add unit tests for the new feature. 
- Update `requests` to `2.31.0`.
- Update `cryptography` to `41.0.4`.
- Add support for Python 3.11 (super inits, imports with try, `tox` additions etc.).
- Update `dev-requirements`:
    - Update `pytest` to the latest `7.4.2`.
    - Update `pylint` to `2.17.7` (`3.0.2` does not support older versions of Python).
- Remove the `--force` CLI option that was deprecated already.

### CODE REVIEW

**Comment by jenia81** [comment]
@JanneKiiskila Please check the code scanning errors
The logs are only for the certificate path, not its content
And I wouldn't drop them since it helps with debugging

File: manifesttool/dev_tool/actions/init.py [hunk 1]
```diff
@@ -200,141 +210,137 @@
...
+        logger.info("imported %s to %s", origin_key_file, dest_key_file)
 
     if origin_cert_file != dest_cert_file:
         shutil.copy(origin_cert_file, dest_cert_file)
-        logger.info(
-            'imported %s to %s',
-            origin_cert_file,
-            dest_cert_file
-        )
+        logger.info("imported %s to %s", origin_cert_file, dest_cert_file)
```
[the remark this answers is not in the document]
Comment (JanneKiiskila) [reply manifesttool/dev_tool/actions/init.py:245]: Logs only file name, not content. OK.

[the remark this answers is not in the document]
Comment (JanneKiiskila) [reply manifesttool/dev_tool/actions/init.py:245]: Logs only file name, not content. OK.

File: manifesttool/dev_tool/actions/init.py [hunk 2]
```diff
@@ -200,141 +210,137 @@
...
-        logger.info('created %s', key_file)
+        logger.info("created %s", key_file)
 
         cert_file.parent.mkdir(parents=True, exist_ok=True)
-        cert_file.write_bytes(
-            cert.public_bytes(serialization.Encoding.DER))
+        cert_file.write_bytes(cert.public_bytes(serialization.Encoding.DER))
 
-        logger.info('created %s', cert_file)
+        logger.info("created %s", cert_file)
```
[the remark this answers is not in the document]
Comment (JanneKiiskila) [reply manifesttool/dev_tool/actions/init.py:340]: Logs only file name, not content. OK.

File: manifesttool/mtool/actions/create.py [hunk 3]
```diff
@@ -35,104 +38,153 @@
...
+                infile = f.name
+                f.write(der_manifest)
+                f.flush()
+                logger.debug("Temporary manifest file: %s", infile)
+            outfile = None
+            with tempfile.NamedTemporaryFile(delete=False) as f:
+                outfile = f.name
+                logger.debug("Temporary signature file: %s", outfile)
+            cmd = [signing_tool_path, digest_algo, key_id, infile, outfile]
+            logger.info("Running %s to sign manifest.", (" ".join(cmd)))
```
[the remark this answers is not in the document]
Comment (JanneKiiskila) [reply manifesttool/mtool/actions/create.py:68]: Logs only file name, not content. OK.

**Comment by JanneKiiskila** [comment]
@jenia81 - cleared false positives with comments. Resolved comments.

File: .github/workflows/pr-checker.yml [hunk 4]
```diff
@@ -1,20 +1,30 @@
 name: PR-checker
-on: [push]
+on: 
+  push:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 */3 * *' # Run every 3 days
```
Comment (JanneKiiskila) [inline .github/workflows/pr-checker.yml:6]: I would drop the cronjob from here.

File: .github/workflows/static-checks.yml [hunk 5]
```diff
@@ -0,0 +1,84 @@
+name: Static checks - misspell and shellcheck
+on: 
+  push:
+  workflow_dispatch:
+
+  schedule:
+    - cron: '0 0 */3 * *' # Run every 3 days
```
Comment (JanneKiiskila) [inline .github/workflows/static-checks.yml:7]: Same here, drop the cronjob from this repo.

**Review by JanneKiiskila — CHANGES_REQUESTED** [review]
Let's drop the cronjobs from this repo.

**Comment by jenia81** [comment]
@JanneKiiskila dropped the cron job from the GitHub action
It's a bit tricky to add the condition to the workflow
I can add it before the steps like this:
`if: github.repository == 'PelionIoT/manifest-tool' && github.event.schedule != '0 0 */3 * *' `
But then the job will start its execution so the machine will be allocated, just the execution time will be shorter.
The only way to make sure that it won't run at all is to create a difference between the workflow files in internal and public repos

**Review by JanneKiiskila — APPROVED** [review]

**Review by JanneKiiskila — APPROVED** [review]

**Review by JanneKiiskila — APPROVED** [review]

**Pull request merged**